src/sas/qtgui/Utilities/GuiUtils.py

```diff
@@ -411,6 +411,7 @@ def createModelItemWithPlot(update_data, name=""):
     Adds 'update_data' to that row.
     """
     py_update_data = update_data
+    py_update_data.name = name # name must match title due to how plots are referenced
```

Comment:
As mentioned during the call, this updates `name` attribute on all plots generated using this utility call.
Those live in:
- Data Operation Utility panel
- Generic Scattering calculator
- Fitting widget
- Slicer parameters widget
- Image viewer
- File loader

I **think** this change is safe, but it would probably be prudent to assure the above functionality remains unchanged (including unit tests).

Reply:
I have performed an initial test of the functionality compared to version 5.0.4:

1) The Data Operation Utility Panel appears to work as before - datasets can be combined and then plotted or passed to the fitting panel. 
I have noticed a couple of things in version 5.0.4 that feel odd, I'm not sure of the intended functionality but these are potentially new tickets:
    - The title of the combined data in the data explorer does not match its name but the name of the second dataset in the combination:
![image](https://user-images.githubusercontent.com/57537928/136011564-e57c198c-dc28-45be-8cc0-017cbe66bc13.png)
   - After deleting one of the generated datasets e.g. MyNewDataName above using the 'Delete Data' button in the data explorer, the name still cannot be re-used in the Data Operation Panel - it is highlighted with a red background as still existing.

2) The generic scattering calculator functionality is changed in this ticket as expected. Plots can now be generated in the fitting calculator after creation by the scattering calculator.

3) The fitting widget uses this function only when creating a new theory index. While I haven't tested all possible routes into this function the basic use of creating a theory from a model appears to work correctly - I can plot it and append it to plots.
   - I notice that in 5.0.4 when saving a project and re-opening it all theories become standard datasets. Again I'm not sure of the intended functionality but this seems odd - should it become a ticket?

4) The slicer parameters widget appears to work as in 5.0.4

5) The Image Viewer appears to have the same issue as the generic scattering calculator in 5.0.4: Once an image has been loaded as data - it cannot be plotted by the fitting panel. This is also fixed by the change in this PR. Another potential issue in 5.0.4 which this PR does not address:
   - The title of the loaded image data is not displayed in the data explorer
![image](https://user-images.githubusercontent.com/57537928/136013582-fc8bb2ce-b74c-4ad7-a3af-32e13b4e3f65.png)

6) The FileLoader uses this method in two places.
    a) DataExplorer.py > updateModelFromData() which is fundamentally used when loading in projects. I can save a project made in this branch and reload it as expected. In sasview 5.0.4 saving and reloading a project would fix any issues of plots not be able to be viewed from the fitting calculator - although at the cost of some datasets being renamed to generic names such as `Data1D` and `Data1D [1]`. When re-opening 5.0.4 projects in this branch the projects load as they would have in 5.0.4.
    b) DataExplorer.py > freezeDataToItem(). I cannot find where this method is being used - but given it passes in a `new_name` argument as a new identifier I see no reason that this would cause issues.
 
I am not confident that I know how the code for these widgets works well enough to write unit tests for them, and now that university has restarted I doubt I will have the time to learn this many different areas of the codebase; hopefully this is a reasonable starting point.